### PR TITLE
682 fargo south university

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: https://github.com/episphere/connect/issues/862

-Add Fargo South University to mapping and dropdown options.
-Show local time instead of UTC time on in-app reports.
-Fix JSON bug found in Check-Out Reports (improper handling of string data with apostrophes).

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -536,36 +536,49 @@ export const addGoToSpecimenLinkEvent = () => {
...
+                    }
+                    checkOutFlag === true ? location.reload() : goToParticipantSearch(); 
+                }, 1500);
+            } else {
+                const visitConcept = document.getElementById('visit-select').value;
+                for(const visit of visitType) {
+                    if (data[conceptIds.collection.selectedVisit] && data[conceptIds.collection.selectedVisit][visit.concept]) {
+                        const visitTime = new Date(data[conceptIds.collection.selectedVisit][visit.concept][conceptIds.checkInDateTime]);
+                        const now = new Date(); 
+                        if(now.getYear() == visitTime.getYear() && now.getMonth() == visitTime.getMonth() && now.getDate() == visitTime.getDate()) {
```
Comment (Gbarra9) [inline src/events.js:567]:
```suggestion
                        if (now.getYear() == visitTime.getYear() && now.getMonth() == visitTime.getMonth() && now.getDate() == visitTime.getDate()) {
```

Comment (Gbarra9) [inline src/events.js:567]:
```suggestion
                        if (now.getYear() == visitTime.getYear() && now.getMonth() == visitTime.getMonth() && now.getDate() == visitTime.getDate()) {
```

**Review by Gbarra9 — APPROVED** [review]
Looks good.

File: src/events.js [hunk 2]
```diff
@@ -536,36 +536,49 @@ export const addGoToSpecimenLinkEvent = () => {
...
-                        if (!confirmContinueCheckIn) return;
+        try {
+            const btnCheckIn = document.getElementById('checkInComplete');
+            btnCheckIn.disabled = true;
+            
+            let query = `connectId=${parseInt(form.dataset.connectId)}`;
+            
+            const response = await findParticipant(query);
+            const data = response.data[0];
+            if(isCheckedIn) {
```
[the remark this answers is not in the document]
Comment (JoeArmani) [reply src/events.js:549]: updated

File: src/events.js [hunk 3]
```diff
@@ -536,36 +536,49 @@ export const addGoToSpecimenLinkEvent = () => {
...
+                setTimeout(() => {
+                    const closeButton = document.querySelector('#biospecimenModal .btn[data-dismiss="modal"]');
+                    if (closeButton) {
+                        closeButton.click();
+                    }
+                    checkOutFlag === true ? location.reload() : goToParticipantSearch(); 
+                }, 1500);
+            } else {
+                const visitConcept = document.getElementById('visit-select').value;
+                for(const visit of visitType) {
```
[the remark this answers is not in the document]
Comment (JoeArmani) [reply src/events.js:563]: updated

File: src/events.js (hunk 1 above)
Comment (JoeArmani) [reply src/events.js:567]: updated

Comment (JoeArmani) [reply src/events.js:567]: updated

File: src/events.js [hunk 4]
```diff
@@ -536,36 +536,49 @@ export const addGoToSpecimenLinkEvent = () => {
...
+                        }
                     }
                 }
+    
+                await handleCheckInModal(data, visitConcept, query);
             }
-
-            await handleCheckInModal(data, visitConcept, query);
+        } catch (error) {
+            const bodyMessage = isCheckedIn ? 'There was an error checking out the participant. Please try again.' : 'There was an error checking in the participant. Please try again.';
```
Comment (anthonypetersen) [inline src/events.js:580]: nice addition

**Review by anthonypetersen — APPROVED** [review]
like the added error handling

**Pull request merged**